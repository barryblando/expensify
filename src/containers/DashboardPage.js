import React from 'react';
import { connect } from 'react-redux';
import Chart from '../components/UI/Chart/Chart';

const DashboardPage = ({ expenses, incomes }) => (
  <div>
    <Chart expenses={expenses} incomes={incomes} />
  </div>
);

const mapStateToProps = state => ({
  expenses: state.expenses,
  incomes: state.incomes,
});

export default connect(mapStateToProps)(DashboardPage);
