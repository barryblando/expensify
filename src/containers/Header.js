import React, { Component } from 'react';
// import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../redux/actions/auth';

import ToolBar from '../components/ToolBar/ToolBar';
import SideDrawer from '../components/SideDrawer/SideDrawer';
import BackDrop from '../components/BackDrop/BackDrop';

export class Header extends Component {
  state = {
    sideDrawerOpen: false,
  };

  // -- Handle hamburger menu open SideDrawer --
  drawerToggleClickHandler = () => {
    // good practice use prevState when toggling state
    this.setState(prevState => ({
      sideDrawerOpen: !prevState.sideDrawerOpen,
    }));
  };

  // -- Handle backdrop close SideDrawer --
  backdropClickHandler = () => {
    this.setState({ sideDrawerOpen: false });
  };

  render() {
    const { startLogout: logout } = this.props;
    const { sideDrawerOpen } = this.state;

    let backdrop;

    // if true then add BackDrop behind SideDrawer
    if (sideDrawerOpen) {
      backdrop = <BackDrop closeHandler={this.backdropClickHandler} />;
    }

    return (
      <header className="toolbar">
        <div className="content-container">
          <ToolBar drawerClickHandler={this.drawerToggleClickHandler} logoutClickHandler={logout} />
          <SideDrawer logoutClickHandler={logout} show={sideDrawerOpen} />
          {backdrop}
        </div>
      </header>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout()),
});

export default connect(
  undefined,
  mapDispatchToProps
)(Header);
