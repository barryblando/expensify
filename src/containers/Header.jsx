import React, { Component } from 'react';
// import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../redux/actions/auth';

import ToolBar from '../components/Toolbar/ToolBar';
import SideDrawer from '../components/SideDrawer/SideDrawer';
import BackDrop from '../components/Backdrop/BackDrop';

export class Header extends Component {
  state = {
    sideDrawerOpen: false,
  };

  drawerToggleClickHandler = () => {
    // good practice use prevState when toggling state
    this.setState(prevState => ({
      sideDrawerOpen: !prevState.sideDrawerOpen,
    }));
  };

  backdropClickHandler = () => {
    this.setState({ sideDrawerOpen: false });
  };

  render() {
    const { startLogout: logout } = this.props;
    const { sideDrawerOpen } = this.state;

    let backdrop;

    if (sideDrawerOpen) {
      backdrop = <BackDrop closeHandler={this.backdropClickHandler} />;
    }

    return (
      <header className="toolbar">
        <div className="content-container">
          <ToolBar drawerClickHandler={this.drawerToggleClickHandler} logoutClickHandler={logout} />
          <SideDrawer logoutClickHandler={logout} show={sideDrawerOpen} />
          {backdrop}
        </div>
      </header>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout()),
});

export default connect(
  undefined,
  mapDispatchToProps
)(Header);
