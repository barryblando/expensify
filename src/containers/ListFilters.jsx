import React, { Component } from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from '../redux/actions/filters';
import { START_DATE, END_DATE } from '../constants';

// eslint-disable-next-line
export class ListFilters extends Component {
  state = {
    calendarFocused: null,
  };

  // ------------------------------------
  // ON DATES CHANGE HANDLER
  // ------------------------------------
  onDatesChange = ({ startDate, endDate }) => {
    const { changeStartDate, changeEndDate } = this.props;
    changeStartDate(startDate);
    changeEndDate(endDate);
  };

  // ------------------------------------
  // ON FOCUS CHANGE HANDLER
  // ------------------------------------
  onFocusChange = calendarFocused => {
    this.setState(() => ({ calendarFocused }));
  };

  // ------------------------------------
  // ON TEXT CHANGE HANDLER
  // ------------------------------------
  onTextChange = e => {
    const { changeTextFilter } = this.props;
    changeTextFilter(e.target.value);
  };

  // ------------------------------------
  // ON SORT CHANGE HANDLER
  // ------------------------------------
  onSortChange = e => {
    const { sortDate, sortAmount } = this.props;
    if (e.target.value === 'date') {
      sortDate();
    } else if (e.target.value === 'amount') {
      sortAmount();
    }
  };

  render() {
    const { startDateId, endDateId, filters } = this.props;
    const { calendarFocused } = this.state;
    return (
      <div className="content-container">
        {/* CONTROLLED INPUTS */}
        <div className="input-group">
          <div className="input-group__item">
            <input
              type="text"
              className="text-input"
              placeholder="Search"
              value={filters.text}
              onChange={this.onTextChange}
            />
          </div>
          <div className="input-group__item">
            <select className="select" value={filters.sortBy} onChange={this.onSortChange}>
              <option value="date">Date</option>
              <option value="amount">Amount</option>
            </select>
          </div>
          <div className="input-group__item">
            <DateRangePicker
              startDate={filters.startDate}
              startDateId={startDateId}
              endDate={filters.endDate}
              endDateId={endDateId}
              onDatesChange={this.onDatesChange}
              focusedInput={calendarFocused}
              onFocusChange={this.onFocusChange}
              showClearDates
              numberOfMonths={1}
              isOutsideRange={() => false}
            />
          </div>
        </div>
      </div>
    );
  }
}

const defaultProps = {
  startDateId: START_DATE,
  endDateId: END_DATE,
};

ListFilters.defaultProps = defaultProps;

// Access state filters
const mapStateToProps = state => ({
  filters: state.filters,
});

// Pass this dispatchers as props
const mapDispatchToProps = dispatch => ({
  changeTextFilter: text => dispatch(setTextFilter(text)),
  sortDate: () => dispatch(sortByDate()),
  sortAmount: () => dispatch(sortByAmount()),
  changeStartDate: startDate => dispatch(setStartDate(startDate)),
  changeEndDate: endDate => dispatch(setEndDate(endDate)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ListFilters);
