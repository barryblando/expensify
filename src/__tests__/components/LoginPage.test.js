import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../containers/LoginPage';

test('should render Login Page correctly', () => {
  const wrapper = shallow(<LoginPage />);
  expect(wrapper).toMatchSnapshot();
});

test('should call startLogin on button click', () => {
  const startLogin = jest.fn(); // fn spy
  const wrapper = shallow(<LoginPage startLogin={startLogin} />);
  wrapper.find('button').simulate('click');
  expect(startLogin).toHaveBeenCalled();
});
