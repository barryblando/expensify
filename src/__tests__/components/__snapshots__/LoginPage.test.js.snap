// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 404 correctly 1`] = `
<div>
  <button>
    Login
  </button>
</div>
`;
